modules/tinymce/src/themes/silver/main/ts/ui/view/ViewWrapper.ts

```diff
@@ -0,0 +1,180 @@
+  },
+  components: makeViews(parts, viewConfigs, providers),
+  slotBehaviours: SimpleBehaviours.unnamedEvents([
+    AlloyEvents.runOnAttached((slotContainer) => SlotContainer.hideAllSlots(slotContainer))
+  ])
+}));
+
+const getCurrentName = (slotContainer: AlloyComponent) => {
+  return Arr.find(SlotContainer.getSlotNames(slotContainer), (name) =>
+    SlotContainer.isShowing(slotContainer, name)
+  );
+};
+
+const hideContainer = (comp: AlloyComponent) => {
+  const element = comp.element;
+  Css.set(element, 'display', 'none');
+  Attribute.set(element, 'aria-hidden', 'true');
+};
+
+const showContainer = (comp: AlloyComponent) => {
+  const element = comp.element;
+  Css.remove(element, 'display');
+  Attribute.remove(element, 'aria-hidden');
+};
+
+const makeViewInstanceApi = (slot: HTMLElement): BridgeView.ViewInstanceApi => ({
+  getContainer: Fun.constant(slot)
+});
+
+const runOnShow = (slotContainer: AlloyComponent, name: string) => {
+  SlotContainer.getSlot(slotContainer, name).each((view) => {
+    View.getPane(view).each((pane) => {
+      const onShow = View.getOnShow(view);
+      onShow(makeViewInstanceApi(pane.element.dom));
+    });
+  });
+};
+
+const runOnHide = (slotContainer: AlloyComponent, name: string) => {
+  SlotContainer.getSlot(slotContainer, name).each((view) => {
+    View.getPane(view).each((pane) => {
+      const onHide = View.getOnHide(view);
+      onHide(makeViewInstanceApi(pane.element.dom));
+    });
+  });
+};
+
+interface SilverViewWrapperSpec extends Sketcher.SingleSketchSpec {
+  backstage: UiFactoryBackstage;
+}
+
+interface SilverViewWrapperDetail extends Sketcher.SingleSketchDetail {
+  uid: string;
+  backstage: UiFactoryBackstage;
+}
+
+interface SilverViewWrapperApis {
+  setViews: (comp: AlloyComponent, viewConfigs: ViewConfig) => void;
+  whichView: (comp: AlloyComponent) => Optional<string>;
+  toggleView: (comp: AlloyComponent, showMainView: () => void, hideMainView: () => void, name: string) => boolean;
+}
+
+const factory: UiSketcher.SingleSketchFactory<SilverViewWrapperDetail, SilverViewWrapperSpec> = (detail, spec) => {
+  const setViews = (comp: AlloyComponent, viewConfigs: ViewConfig) => {
+    Replacing.set(comp, [ makeSlotContainer(viewConfigs, spec.backstage.shared.providers) ]);
+  };
+
+  const whichView = (comp: AlloyComponent): Optional<string> => {
+    return Composing.getCurrent(comp).bind(getCurrentName);
+  };
+
+  const toggleView = (comp: AlloyComponent, showMainView: () => void, hideMainView: () => void, name: string): boolean => {
+    return Composing.getCurrent(comp).exists((slotContainer) => {
+      const optCurrentSlotName = getCurrentName(slotContainer);
+      const hasSameName = optCurrentSlotName.exists((current) => name === current);
+      const exists = SlotContainer.getSlot(slotContainer, name).isSome();
+
+      if (exists) {
+        SlotContainer.hideAllSlots(slotContainer);
+
+        if (!hasSameName) {
```

Comment:
`hasSameName` is what the predicate is representing, but conceptually it is representing "am I toggling this on" or "am I toggling this off". Should we use a predicate name that reflects that more?

Reply:
Changed this to `isTogglingCurrentView` not sure if that is more clear.